### PR TITLE
[3.9] Log the event on actionlogs when Joomla update

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #30074 .

### Summary of Changes
Log the Joomla update event on the Actionlogs

### Testing Instructions
joomlaupdate -> options
change  to Custom URL
and put this manifest https://ci.joomla.org/artifacts/joomla/joomla-cms/staging/30157/downloads/34205/pr_list.xml
you'll find it here on the download pkg link  Prebuilt packages are available for download. 

### Actual result BEFORE applying this Pull Request
no event is logged when joomla update

### Expected result AFTER applying this Pull Request
the event is logged
Screenshot from 2020-07-21 19-39-11

### CODE REVIEW

File: plugins/actionlog/joomla/joomla.php [hunk 1]
```diff
@@ -1042,4 +1042,36 @@ public function onAfterPurge($group = 'all')
...
+	 *
+	 * @since   3.9.21
+	 */
+	public function onAfterUpdate($version)
+	{
+		$context = $this->app->input->get('option');
+		$user    = JFactory::getUser();
+
+
+
```
Comment (Quy) [inline plugins/actionlog/joomla/joomla.php:1063]:
```suggestion
```

Comment (Quy) [inline plugins/actionlog/joomla/joomla.php:1063]:
```suggestion

```

Comment (Quy) [inline plugins/actionlog/joomla/joomla.php:1063]:
```suggestion
		$user    = JFactory::getUser();
```

File: plugins/actionlog/joomla/joomla.php [hunk 2]
```diff
@@ -1042,4 +1042,36 @@ public function onAfterPurge($group = 'all')
...
+		$message = array(
```
Comment (Quy) [inline plugins/actionlog/joomla/joomla.php]:
```suggestion
		$user    = JFactory::getUser();
		$message = array(
```

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 7b32ca93a2f636df50f65e16af0650e24ac8a1b1<br /><br />Got: User xyz updated the CMS to 3.9.21-dev+pr.30157 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30157">issues.joomla.org/tracker/joomla-cms/30157</a>.</sub>

**Comment by SharkyKZ** [comment]
There are already `onJoomlaBeforeUpdate` and `onJoomlaAfterUpdate` events in 4.0.

**Comment by alikon** [comment]
but they are used just for plugin installer/override
so what we can do ?

**Comment by LukeDouglas2** [comment]
Thanks mates!

**Comment by SharkyKZ** [comment]
> 
> 
> but they are used just for plugin installer/override
> so what we can do ?

They can be used by any plugin.

File: administrator/language/en-GB/en-GB.plg_actionlog_joomla.ini [hunk 3]
```diff
@@ -46,6 +46,7 @@ PLG_ACTIONLOG_JOOMLA_USER_REGISTERED="User <a href='{accountlink}'>{username}</a
 PLG_ACTIONLOG_JOOMLA_USER_REMIND="User <a href='{accountlink}'>{username}</a> requested a username reminder for their account"
 PLG_ACTIONLOG_JOOMLA_USER_RESET_COMPLETE="User <a href='{accountlink}'>{username}</a> completed the password reset for their account"
 PLG_ACTIONLOG_JOOMLA_USER_RESET_REQUEST="User <a href='{accountlink}'>{username}</a> requested a password reset for their account"
+PLG_ACTIONLOG_JOOMLA_USER_UPDATE="User <a href='{accountlink}'>{username}</a> updated the CMS to {version}"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_actionlog_joomla.ini]:
```suggestion
PLG_ACTIONLOG_JOOMLA_USER_UPDATE="User <a href='{accountlink}'>{username}</a> updated Joomla to {version}"
```

**Comment by ChristineWk** [comment]
Made as in the first test on another Site. (J!3.9.21-dev+pr.30157 was available). 
But got: Download failed
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30157">issues.joomla.org/tracker/joomla-cms/30157</a>.</sub>

**Comment by alikon** [comment]
as suggested https://github.com/joomla/joomla-cms/pull/30157#issuecomment-662034514 in j4 there is already such event `onJoomlaAfterUpdate`

made the relative changes

please re-test

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 19414dc70e569fcdd0350c40ae4ad77487639e22<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30157">issues.joomla.org/tracker/joomla-cms/30157</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 19414dc70e569fcdd0350c40ae4ad77487639e22<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30157">issues.joomla.org/tracker/joomla-cms/30157</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30157">issues.joomla.org/tracker/joomla-cms/30157</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**